pkg/interface/chat/src/js/components/lib/invite-element.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import { Sigil } from '/components/lib/icons/sigil';
+import { deSig } from '/lib/util';
+import urbitOb from 'urbit-ob';
+
+
+export class InviteElement extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      members: '',
+      error: false,
+      success: false
+    };
+  }
+
+  modifyMembers() {
+    const { props, state } = this;
+
+    let aud = [];
+    let isValid = true;
+    if (state.members.length > 2) {
+      aud = state.members.split(',')
+        .map((mem) => mem.trim())
+        .map(deSig)
+        .map((mem) => {
+          return `~${mem}`;
+        });
```

Comment:
I'm not sure if a JS runtime will optimise this away, but it may be more efficient to move from:

```
foo.map(f).map(g).map(h)
```

to

```
foo.map(x => h(g(f(x))))
```

Additionally, you don't need the braces and `return` in the last `.map`.  You can just do:

```
.map(mem => `~${mem}`)
```

Reply:
Seems cleaner, will try it

Reply:
looks nasty but it works?